### DISCUSSION TITLE
Is there a reason this implementation was chosen?

### DISCUSSION THREAD

**Discussion by M15terHyde** [aiortc/aioquic#307] [Q&A] [answered]
https://github.com/aiortc/aioquic/blob/7b3f24bbdec8b93dfee0bad954f493ea7767241f/src/aioquic/tls.py#L1223
I understand it's shifting each byte onto the right to recreate the integer of the message_length but why reinvent the wheel when python integers already have int.from_bytes.
Was there a reason:
```
message_length = 0
message_length.from_bytes( self._receive_buffer[1:4], byteorder='big' )
```
was not chosen?

**Comment by jlaine** [comment, marked as answer]
Hah good catch, I'm guessing it's just me not knowing the standard library inside out!

```python
message_length = 4 + int.from_bytes(buffer[1:4], byteorder="big")
```

Here is a PR which makes use of int.from_bytes and int.to_bytes:

https://github.com/aiortc/aioquic/pull/312
